external/storm-kafka-client/src/main/java/org/apache/storm/kafka/trident/TridentKafkaState.java

```diff
@@ -98,14 +107,19 @@ public void updateState(List<TridentTuple> tuples, TridentCollector collector) {
             }
 
             if(exceptions.size() > 0){
-                String errorMsg = "Could not retrieve result for messages " + tuples + " from topic = " + topic 
-                        + " because of the following exceptions: \n";
-                for (ExecutionException exception : exceptions) {
-                    errorMsg = errorMsg + exception.getMessage() + "\n";
-                }
-                LOG.error(errorMsg);
-                throw new FailedException(errorMsg);
-            }
+		StringBuilder errorMsg = new StringBuilder("Could not retrieve result for messages ").append(tuples)
+			.append(" from topic = ").append(topic).append(" because of the following exceptions: ").append(System.lineSeparator());
+				
+		for (ExecutionException exception : exceptions) {
+			errorMsg = errorMsg.append(exception.getMessage()).append(System.lineSeparator()); ;
+		}
+		String message = errorMsg.toString();
+		LOG.error(message);
+		throw new FailedException(message);
+	    }
+	    long latestTime=System.currentTimeMillis();
+	    LOG.info("Emitted record {} sucessfully in {} ms to topic {} ", new Object[]{emittedRecords,latestTime-currentTime,topic});
```

Comment:
Please keep log simple for INFO, since it will be called for every updates. Just leaving record count is enough.
If you really want to leave these records, this should be DEBUG.